pkg/registry/registry_docker.go

```diff
@@ -0,0 +1,151 @@
+package registry
+
+import (
+	"bufio"
+	"context"
+	"encoding/base64"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"strings"
+	"text/template"
+
+	"github.com/docker/docker/api/types"
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/retagger/pkg/images"
+)
+
+func (r *Registry) Login() error {
+	r.logger.Log("level", "debug", "message", fmt.Sprintf("logging in to %s registry", r.host))
+
+	authConfig := types.AuthConfig{
+		Username:      r.username,
+		Password:      r.password,
+		ServerAddress: r.host,
+	}
+	res, err := r.docker.RegistryLogin(context.Background(), authConfig)
+	if err != nil {
+		return microerror.Maskf(err, "could not login to registry")
+	} else if res.Status != "Login Succeeded" {
+		return microerror.Mask(dockerError)
+	}
+
+	r.registryAuth = res.IdentityToken
+
+	return nil
+}
+
+func (r *Registry) PullImage(image string, sha string) error {
+	if sha == "" {
+		return microerror.Maskf(invalidArgumentError, "%s SHA should not be empty", image)
```

Comment:
Why is there a need to check the sha but not the image?

Reply:
because legacy :)